src/behaviors/emqttb_behavior_pub.erl

```diff
@@ -75,6 +88,15 @@ terminate(_Shared, Conn) ->
 %% Internal functions
 %%================================================================================
 
+error_fun(SetLatencyKey, Group) ->
+  SetLatency = ?CFG(SetLatencyKey),
+  AvgWindow = 250,
+  (emqttb_metrics:get_rolling_average(?GROUP_OP_TIME(Group, ?AVG_PUB_TIME), AvgWindow) -
+     erlang:convert_time_unit(SetLatency, millisecond, microsecond)).
```

Comment:
Q: usually the error is `Target - Current`, so that the PID output is "positive" when `Current < Target`.  Is there a particular reason to be `Current - Target` here?

Reply:
The dependency of produce interval on latency is assumed to be inverse (lesser interval -> more often messages get send -> more latency), so the control must be reversed.